File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AnnoyedQue.java
package Auxiliary_Modules;

public class AnnoyedQue {
    private final RefreshQ q1 = new RefreshQ();
    private final RefreshQ q2 = new RefreshQ();

    private final RefreshQ stuffedQue = new RefreshQ();

    public AnnoyedQue(int queLim) {
        q1.setLimit(queLim);
        q2.setLimit(queLim);
        stuffedQue.setLimit(queLim);
    }
    public void learn(String ear){
        if (q1.contains(ear)){
            q2.add(ear);
            stuffedQue.stuff(ear);
            return;
        }
        q1.add(ear);
    }
    public boolean isAnnoyed(String ear){
        return q2.strContainsResponse(ear);
    }
    public void reset() {
        // Insert unique throwaway strings to reset the state
        for (int i = 0; i < this.q1.getLimit(); i++) {
            this.learn("throwaway_string_" + i);
        }
    }
    public boolean AnnoyedLevel(String ear, int level) {
        int count = 0;
        for (String item : this.stuffedQue.elements) {
            if (item.equals(ear)) {
                count++;
            }
        }
        return count > level;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AnnoyedQue.kt
package auxiliary_modules

class AnnoyedQue(queLim: Int) {
    private val q1 = RefreshQ()
    private val q2 = RefreshQ()
    private val stuffedQue = RefreshQ()

    init {
        q1.limit = queLim
        q2.limit = queLim
        stuffedQue.limit = queLim
    }

    fun learn(ear: String) {
        if (q1.contains(ear)) {
            q2.add(ear)
            stuffedQue.stuff(ear)
            return
        }
        q1.add(ear)
    }

    fun isAnnoyed(ear: String?): Boolean {
        return q2.strContainsResponse(ear!!)
    }

    fun reset() {
        // Insert unique throwaway strings to reset the state
        for (i in 0 until q1.limit) {
            learn("throwaway_string_$i")
        }
    }

    fun AnnoyedLevel(ear: String, level: Int): Boolean {
        var count = 0
        for (item in stuffedQue.elements) {
            if (item == ear) {
                count++
            }
        }
        return count > level
    }
}